src/text-editor-presenter.coffee

```diff
@@ -595,33 +641,24 @@ class TextEditorPresenter
       isVisible = isVisible and @showLineNumbers
     isVisible
 
-  updateLineNumbersState: (tileState, startRow, endRow) ->
-    tileState.lineNumbers ?= {}
-    visibleLineNumberIds = {}
+  isSoftWrappedRow: (bufferRow, screenRow) ->
+    return false if screenRow is 0
 
-    if startRow > 0
-      rowBeforeStartRow = startRow - 1
-      lastBufferRow = @model.bufferRowForScreenRow(rowBeforeStartRow)
-    else
-      lastBufferRow = null
+    @model.bufferRowForScreenRow(screenRow - 1) is bufferRow
```

Comment:
@nathansobo: I feel like we should be able to optimize this. Instead of converting screen rows, do you think we could cache this information into `TokenizedLine` when we soft wrap a line in `DisplayBuffer`? This call is quite expensive in terms of performance, and it grows as the file gets longer.

Until we implement a faster `RowMap` I feel like we should be restrictive about calling those methods.